config/filter.d/cyrus-imap.conf

```diff
@@ -13,7 +13,7 @@ before = common.conf
 
 _daemon = (?:cyrus/)?(?:imap(d|s)?|pop3(d|s)?)
 
-failregex = ^%(__prefix_line)sbadlogin: \S+ ?\[<HOST>\] \S+ .*?\[?SASL\(-13\): (authentication failure|user not found): .*\]?$
+failregex = ^%(__prefix_line)sbadlogin: (\S+)? ?\[<HOST>\] \S+ .*?\[?SASL\(-13\): (authentication failure|user not found): .*\]?$
```

Comment:
1. `(?:...)` - no capturing
2. Definitelly no, because as I already wrote, `\]?` is optional, so expression `.*\]?$` matchs always, no matter what it is
```python
re.search(r"""(.*\]?$)""", r"""test""").groups()
('test',)
```